CMakeLists.txt

```diff
@@ -14,6 +14,37 @@ set(INSTALL_CMAKE_DIR lib/cmake CACHE PATH "Installation directory for cmake fil
 
 set(CMAKE_FIND_PACKAGE_PREFER_CONFIG TRUE)
 
+set(CTEST_MEMORYCHECK_TYPE "Valgrind")
```

Comment:
I probably wouldn't have this at the top of the CMakeLists.txt, since it's relatively unimportant/specific. It probably makes more sense in the test section?

Reply:
I moved this part to the section of `BUILD_TESTING`. I put above the section where it is about to do either TSAN or ASAN check. CTEST memcheck uses Valgrind by default, but I want to make sure that Valgrind is ran, instead of other options. According to the CTEST documentations, there are other options like "Valgrind, Purify, BoundsChecker, DrMemory, CudaSanitizer, ThreadSanitizer, AddressSanitizer, LeakSanitizer, MemorySanitizer and UndefinedBehaviorSanitizer."
https://cmake.org/cmake/help/latest/variable/CTEST_MEMORYCHECK_TYPE.html